src/bitcoin/wallet.rs

```diff
@@ -61,14 +63,21 @@ impl Wallet {
             Ok(WalletInfoResponse {
                 hd_seed_id: None, ..
             }) => {
+                // Rescan wallet to ensure funding is picked up
+                self.bitcoind_client.rescan(&self.name).await?;
+
                 // The wallet may have been previously created, but the `sethdseed` call may have failed
                 let wif = self.seed_as_wif();
 
                 self.bitcoind_client
                     .set_hd_seed(&self.name, Some(true), Some(wif))
                     .await
             }
-            _ => Ok(()),
+            _ => {
+                // Rescan wallet to ensure funding is picked up
+                self.bitcoind_client.rescan(&self.name).await?;
```

Comment:
This is unnecessary as the wallet would be created on bitcoind already.

Reply:
This is necessary, because otherwise we don't pick up money if the address was funded prior to the wallet being created.
This can happen if I run Nectar to e.g. display the deposit addresses (creates the wallet...) and then fund the address and then run it again against the same node.

Reply:
This matching branch means the wallet was already created on bitcoind. It means nectar is **not** creating the wallet, the wallet was already here.

Please describe a sequence of event that would justify re-scanning here.

If the `rescan` is properly done in the other match branches (when the wallet **is** created by nectar) then it should be enough.

Reply:
You are right, thanks! Somehow I thought I remember that I needed it - but that was probably some weird scenario when testing the new setup.
I just tried it with the current comit-script setup that pre-funds (and then Nectar is started, creates the wallet, rescans - upon starting it again the rescan is not needed).

Removed it!